microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -284,11 +284,21 @@ private void validateRetryAnnotation(NormalAnnotation annotation)
 					? jitterExpr.resolveConstantExpressionValue()
 					: null;
 
-			int delayNum = delayConstant != null ? (int) delayConstant : -1;
-			int maxDurationNum = maxDurationConstant != null
-					? (int) maxDurationConstant
-					: -1;
-			int jitterNum = jitterConstant != null ? (int) jitterConstant : 0;
+			int delayNum = -1;

```

Comment:
Something I missed initially is the type specification for [Retry](https://download.eclipse.org/microprofile/microprofile-fault-tolerance-2.0.2/apidocs/org/eclipse/microprofile/faulttolerance/Retry.html) annotation members. I.e. delay is of type `long`, but we assign it as `int` here. In that case, your case would be using the `Long` wrapper instead. We can change this for consistency. Thoughts @rgrunber?

Reply:
I just tried creating a number that's larger than the int range, and when I do JDT complains `The literal 5000000000 of type int is out of range Java(536871066)` . So this might be a bug in JDT. I think it should be fine for now.

Reply:
After reproducing this, I think I know what JDT expects. Instead of providing a number larger than an `int` (eg. a `long`) as just "the number", it expects it in the form `5000000000L`. Then it won't complain. I guess it needs the user to explicitly mark it as long.

Once that is done, `resolveConstantExpressionValue(..)` will return a `Long`, not an `Integer` so we could either :
- Handle both `Long` and `Integer` as separate cases
- See if you can combine all cases using `Long.valueOf(...)`